### ISSUE TITLE
Change the default behavior of space-before-function-paren with async arrow functions

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#8267]
**What rule do you want to change?**

[`space-before-function-paren`](http://eslint.org/docs/rules/space-before-function-paren)

**Does this change cause the rule to produce more or fewer warnings?**

More

**How will the change be implemented? (New option, new default behavior, etc.)?**

This will change the default of the `asyncArrow` option from `"ignore"` to `"always"`.

As a result, the option will be consistent with all of the rule's other options. When the `asyncArrow` option was added in https://github.com/eslint/eslint/pull/7180, we decided to set the default to `"ignore"` at first and consider changing it in 4.0.

**Please provide some example code that this change will affect:**

```js
/* eslint space-before-function-paren: "error" */
async() => {};
```

**What does the rule currently do for this code?**

It does not report an error.

**What will the rule do after it's changed?**

It will report an error.

**Comment by not-an-aardvark** [comment]
Adding this to the TSC agenda because it's a breaking change.

**Comment by kaicataldo** [comment]
Do we also need issues for other config default changes listed in this comment? https://github.com/eslint/eslint/issues/7407#issuecomment-254930694

**Comment by not-an-aardvark** [comment]
Looking at the issue again, I'm no longer convinced that those other config defaults changes are worth pursuing. I'm fine with considering them if someone else thinks they're worthwhile though.

**Comment by alberto** [comment]
Since we already have a majority in agreement with this, I think we can accept it.
